storage/libradb/src/lib.rs

```diff
@@ -538,14 +539,39 @@ impl LibraDB {
             })
             .collect::<Result<Vec<_>>>()?;
 
+        // TODO: cache last epoch change version to avoid a DB access in most cases.
+        let client_epoch = self.ledger_store.get_epoch(client_known_version)?;
+        let current_epoch = if ledger_info.next_validator_set().is_some() {
+            ledger_info.epoch() + 1
+        } else {
+            ledger_info.epoch()
+        };
+        let validator_change_proof = if client_epoch < current_epoch {
+            let mut ledger_infos = self
+                .ledger_store
+                .get_latest_ledger_infos_per_epoch(client_epoch)?;
+            if ledger_infos
+                .last()
+                .ok_or_else(|| LibraDbError::NotFound(String::from("Latest epoch LedgerInfo")))?
+                .ledger_info()
+                .next_validator_set()
+                .is_none()
+            {
+                ledger_infos.pop();
+            }
+            ledger_infos
```

Comment:
Does this have the same problem? Imagine there is an epoch change that happens right after the `get_latest_ledger_info` call at line 466 and before this piece of code is executed. Then this might return the newly added epoch change, which is not desired?

Reply:
Looks like a later commit fixes this?

Reply:
Yes the third commit resolves this.